src/Analysis/Ast/Impl/Modules/PythonModule.cs

```diff
@@ -235,7 +235,14 @@ internal PythonModule(ModuleCreationOptions creationOptions, IServiceContainer s
         /// <summary>
         /// Returns module content (code).
         /// </summary>
-        public string Content => _buffer.Text;
+        public string Content {
+            get {
+                lock (AnalysisLock) {
```

Comment:
This seems spooky to me since `Reset` uses `Content` under `AnalysisLock`, but I guess if that's how C# locks work...